File: src/main/java/it/giaquinto/springberry/model/sorted/SortedStrategy.java
package it.giaquinto.springberry.model.sorted;

import java.util.Collection;
import java.util.List;

public interface SortedStrategy<T extends Comparable> {
    List<T> sortDescending(final Collection<T> toOrderedDescending);

    List<T> sortGrowing(final Collection<T> toOrderedGrowing);

    T[] sortDescendingArray(final Collection<T> toOrderedDescending);

    T[] sortGrowingArray(final Collection<T> toOrderedGrowing);


}

File: src/main/java/it/giaquinto/springberry/model/sorted/SortedStrategy.kt
package it.giaquinto.springberry.model.sorted

interface SortedStrategy<T : Comparable<*>?> {
    fun sortDescending(toOrderedDescending: Collection<T>): List<T>?
    fun sortGrowing(toOrderedGrowing: Collection<T>): List<T>?
    fun sortDescendingArray(toOrderedDescending: Collection<T>): Array<T>
    fun sortGrowingArray(toOrderedGrowing: Collection<T>): Array<T>
}